### ISSUE TITLE
add numpy and cython to the requirements for the package

### ISSUE THREAD

**Issue by fawzi** [mdtraj/mdtraj#1119]
currently pip install of a requirements file with both numpy cython and mdtraj fails because the mdtraj does not list them as dependencies, so they are installed in parallel and mdtraj fails with a message suggesting to pip install or use conda.
That is better than nothing (and clearly installing them fixes the problem), but I find that adding the correct dependencies to mdtraj would be much better.

**Comment by marscher** [comment]
I added a PR https://github.com/mdtraj/mdtraj/pull/934 for this quite some time ago, but it has been reverted, since the automatic install with easy_install is not as reliable as we want it to be.

However we are using the same pattern in PyEMMA without any (bigger) problems. At least cython can be installed seamlessly during setup phase.

**Comment by fawzi** [comment]
actually scipy is also required...

**Comment by fawzi** [comment]
I see, well yes one can always work around things, still it is annoying, maybe now your patch can be reapplied ? ;)

**Comment by rmcgibbo** [comment]
[This distutils-sig thread](https://mail.python.org/pipermail/distutils-sig/2015-October/027161.html) describes some of the pain that comes from listing numpy as a dependency to setuptools and why most packages don't.

**Comment by marscher** [comment]
So his main point is the recursive upgrade behaviour of pip being hated by a lot of people, right?

**Comment by rmcgibbo** [comment]
I think that's true. We could search back through all of the issues/bugs/prs on this repo too that arose from setuptools/numpy iteractions.

**Comment by gph82** [comment]
Hi, before I open another issue, is this why I'm seeing something like this?
```
conda create --name python3.6_for_ipywidgets6 python=3.6
...
source activate python3.6_for_ipywidgets6
pip install mdtraj 
Collecting mdtraj
  Downloading mdtraj-1.8.0.zip (20.2MB)
    100% |################################| 20.2MB 47kB/s 
    Complete output from command python setup.py egg_info:
    --------------------------------------------------------------------------------
    Error: building mdtraj requires numpy and cython>=0.19
    
    Try running the command ``pip install numpy cython`` or
    ``conda install numpy cython``.
    
    or see http://docs.scipy.org/doc/numpy/user/install.html and
    http://cython.org/ for more information.
    
    If you're feeling lost, we recommend downloading the (free) Anaconda python
    distribution https://www.continuum.io/downloads, because it comes with
    these components included.
    --------------------------------------------------------------------------------
    
    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-build-tmlqsnoy/mdtraj/
```

**Comment by marscher** [comment]
I asked this like 2 years ago. It got accepted and then reverted because 
of "troubles". But maybe somebody changed his/her mind in the meantime :)

**Comment by rmcgibbo** [comment]
Yeah, as you suspected I'm marking this as wont-fix.

**Comment by rmcgibbo** [comment]
This may be possible to implement in an acceptable way once PEP518 support is in pip though.

**Comment by marscher** [comment]
Ah good to know. Thanks for the reference!
